module/core/src/Application/Request/ParamConverter/AggregateRootParamConverter.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Core\Application\Request\ParamConverter;
+
+use Ergonode\EventSourcing\Domain\AbstractAggregateRoot;
+use Ergonode\EventSourcing\Infrastructure\Manager\EventStoreManager;
+use Ergonode\SharedKernel\Domain\AggregateId;
+use Ergonode\Workflow\Domain\Entity\AbstractWorkflow;
+use Ergonode\Workflow\Domain\Provider\WorkflowProvider;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\Assert;
+
+/**
+ */
+class AggregateRootParamConverter implements ParamConverterInterface
+{
+    /**
+     * @var EventStoreManager
+     */
+    private EventStoreManager $manager;
+
+    /**
+     * @var WorkflowProvider
+     */
+    private WorkflowProvider $provider;
+
+    /**
+     * @param EventStoreManager $manager
+     * @param WorkflowProvider  $provider
+     */
+    public function __construct(EventStoreManager $manager, WorkflowProvider $provider)
+    {
+        $this->manager = $manager;
+        $this->provider = $provider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration): void
+    {
+        $name = $configuration->getName();
+        $class = $configuration->getClass();
+        if (null === $name) {
+            throw new BadRequestHttpException('Param Converter argument \'Name\'is missing');
+        }
+
+        if (AbstractWorkflow::class === $class) {
```

Comment:
As it is non-generic behavior this should be replaced with a dedicated converter.